nrf-hal-common/src/pwm.rs

```diff
@@ -0,0 +1,1017 @@
+//! HAL interface to the PWM peripheral.
+//!
+//! The pulse with modulation (PWM) module enables the generation of pulse width modulated signals on GPIO.
+
+use core::sync::atomic::{compiler_fence, Ordering};
+
+#[cfg(any(feature = "52833", feature = "52840"))]
+use crate::{
+    gpio::{Output, Pin, Port, PushPull},
+    pac::{
+        generic::Reg,
+        pwm0::{
+            _EVENTS_LOOPSDONE, _EVENTS_PWMPERIODEND, _EVENTS_SEQEND, _EVENTS_SEQSTARTED,
+            _EVENTS_STOPPED, _TASKS_NEXTSTEP, _TASKS_SEQSTART, _TASKS_STOP,
+        },
+        PWM0, PWM1, PWM2, PWM3,
+    },
+    target_constants::{SRAM_LOWER, SRAM_UPPER},
+    time::*,
+};
+
+/// A safe wrapper around the raw peripheral.
+#[derive(Debug)]
+pub struct Pwm<T: Instance> {
+    pwm: T,
+    duty: [u16; 4],
+}
+
+impl<T> Pwm<T>
+where
+    T: Instance,
+{
+    /// Takes ownership of the peripheral and applies sane defaults.
+    pub fn new(pwm: T) -> Pwm<T> {
+        compiler_fence(Ordering::SeqCst);
+        let duty = [0u16; 4];
+        pwm.enable.write(|w| w.enable().enabled());
+        pwm.mode.write(|w| w.updown().up());
+        pwm.prescaler.write(|w| w.prescaler().div_1());
+        pwm.countertop
+            .write(|w| unsafe { w.countertop().bits(32767) });
+        pwm.loop_.write(|w| w.cnt().disabled());
+        pwm.decoder.write(|w| {
+            w.load().individual();
+            w.mode().refresh_count()
+        });
+        pwm.seq0.refresh.write(|w| unsafe { w.bits(0) });
+        pwm.seq0.enddelay.write(|w| unsafe { w.bits(0) });
+        pwm.seq1.refresh.write(|w| unsafe { w.bits(0) });
+        pwm.seq1.enddelay.write(|w| unsafe { w.bits(0) });
+
+        pwm.seq0
+            .ptr
+            .write(|w| unsafe { w.bits(duty.as_ptr() as u32) });
+        pwm.seq0.cnt.write(|w| unsafe { w.bits(4) });
+        pwm.seq1
+            .ptr
+            .write(|w| unsafe { w.bits(duty.as_ptr() as u32) });
+        pwm.seq1.cnt.write(|w| unsafe { w.bits(4) });
+
+        Self { pwm, duty }
+    }
+
+    /// Sets the PWM clock prescaler.
+    #[inline(always)]
+    pub fn set_prescaler(&self, div: Prescaler) -> &Self {
+        self.pwm.prescaler.write(|w| w.prescaler().bits(div.into()));
+        self
+    }
+
+    /// Sets the PWM clock prescaler.
+    #[inline(always)]
+    pub fn get_prescaler(&self) -> Prescaler {
```

Comment:
The convention for getters is to omit the `get_` prefix, so:

```suggestion
    pub fn prescaler(&self) -> Prescaler {
```

(same for the others)